File: backend/src/main/kotlin/com/moa/backend/model/dto/IdeaDto.kt
package com.moa.backend.model.dto

import com.moa.backend.model.*
import com.moa.backend.model.slim.*
import java.util.*

data class IdeaDto (

    var id: Long = 0,

    var title: String,

    var description: String,

    var owner: UserSlimDto,

    var status: Status,

    var creationDate: Date?,

    var tags: MutableList<TagSlimDto>?,

    var comments: MutableList<CommentSlimDto>?,

    var ideaBox: IdeaBoxSlimDto,

    var likes: MutableList<UserSlimDto>?,

    var requiredJuries: MutableList<UserSlimDto>?,

    var scoreSheets: MutableList<ScoreSheetSlimDto>?,
)

File: frontend/src/app/models/dto/ideaDto.ts
import { Judgement } from '../judgement';
import { CommentSlimDto } from '../slimDto/commentSlimDto';
import { IdeaBoxSlimDto } from '../slimDto/ideaBoxSlimDto';
import { ScoreSheetSlimDto } from '../slimDto/scoreSheetSlimDto';
import { ScoreSlimDto } from '../slimDto/scoreSlimDto';
import { TagSlimDto } from '../slimDto/tagSlimDto';
import { UserSlimDto } from '../slimDto/userSlimDto';

export interface IdeaDto {
  id?: number;
  title?: string;
  description?: string;
  scoreSheets?: ScoreSheetSlimDto[];
  owner?: UserSlimDto;
  status?: string;
  creationDate?: Date;
  tags?: TagSlimDto[];
  comments?: CommentSlimDto[];
  ideaBox?: IdeaBoxSlimDto;
  likes?: UserSlimDto[];
  requiredJuries?: UserSlimDto[];

}
